modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1334 @@
+<p>Try to insert content here</p>
+----
+
+After executing an `onclick` event within the current `selection | node` using `<button onclick="tinymce.activeEditor.insertContent('text')">insertContent</button>`, the output would modify the existing `contenteditable="false";` element, which should not be possible.
+
+[source, HTML]
+----
+<p>Try to insert text content here</p>
+----
+
+The new additions and improvements with the introduction of {productname} 6.4.1, include a new `isEditable` API been added to the `editor.selection` which now blocks the `editor.insertContent` API and `mceInsertContent` command when the contents return a `true` value if the editor detects the element is a `contenteditable="false";` element in a `contenteditable="false";` root.
+
+As a result, any `editor.selection` elements that return a `true` value when a specified element is `contenteditable="true"`, will call the new `isEditable` API to execute the new blocker.
+
+=== New `editor.dom.isEditable` and `editor.selection.isEditable` APIs that returns whether the specified node is editable
+//#TINY-9462
+
+In previous versions of {productname}, elements within a `contenteditable="false"` root could be unexpectedly modified by the `editor.insertContent` API and the `mceInsertContent` command..
+
+As a consequence all elements within the root `body` set to `contenteditable="false";` value, were affected by the issue.
+
+The new additions and improvements with the introduction of {productname} 6.4.1, include a new `isEditable` API been added to the `editor.dom` which now blocks the `editor.insertContent` API and `mceInsertContent` command when the contents return a `true` value if the editor detects the element is a `contenteditable="false";` in a `contenteditable="false";` root.
+
+As a result, any elements that return a `true` value when a specified element is `contenteditable="true"`, will call the new `isEditable` API to action the new blocker to ensure `contenteditable="false";` elements cannot be modified when they are within a root that is set to `contenteditable="false"`.
+
+=== New setText and setIcon methods added to menu button and toolbar button APIs
+//#TINY-9268
+
+In previous versions of {productname}, it was not possible to change the `text` or `icon` of a toolbar button after it had been set for the first time. As a consequence, the menu and toolbar button API, would not extend or update the parent `icon` or `text` directly, based upon what the user had selected within the menubar or toolbar.
+
+In {productname} 6.4.1, two new additions were added to the menu button API.
+
+. Added `setText` and `setIcon` properties to both the toolbar button `api`, and menu button `api`.
+. Provided the menu button `api` property as an *argument* which will fetch a function that will allow menu items to change the `icon` or `text` of the parent menu button.
+
+As a result, in {productname} 6.4.1, menu items can change based on the parent menu button `api` properties.
```

Comment:
```suggestion
In {productname} 6.4, two new methods (`setText` and `setIcon` ) were added to the menu and toolbar button APIs, that now update the parent icon or text based on what menu is selected from the button.

As a result, in {productname} 6.4, menu items can now change based on the parent menu and toolbar button selection.
```

Reply:
@tiny-ben-tran - Moved this suggestion here due to being outdated, can you please review when possible.

Reply:
```suggestion
In previous versions of {productname}, it was not possible to change the `text` or `icon` of a toolbar or menu button dynamically.

 In {productname} 6.4, two new methods `setText` and `setIcon` were added to the menu and toolbar button APIs, to enable updating the text and icon of the buttons after the editor initialization.
```

Reply:
```suggestion
In previous versions of {productname}, it was not possible to change the `text` or `icon` of a toolbar or menu button dynamically.

In {productname} 6.4, two new methods `setText` and `setIcon` were added to the menu and toolbar button APIs, to enable updating the `text` and `icon` of the buttons after the editor initialization.
```